### DISCUSSION TITLE
[❄️ Snowflake Official] Grant Ownership Update

### DISCUSSION THREAD

**Discussion by sfc-gh-jcieslak** [Snowflake-Labs/terraform-provider-snowflake#2235] [Announcements]
Hey 👋 

Following on [our previous grants announcement](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1890), we are in the process of analyzing, designing, and re-designing grants.

We are considering a case that we would like to take a closer look at and gather some feedback, and that would be granting ownership through our provider. It's rather problematic in some cases and we would like to gather some information on what are your use cases for granting ownership to better understand your needs and adjust new resources accordingly to be able to support as many cases as possible.

To give us your feedback, head over to [the form](https://forms.gle/KQUB45GEzkgRTUHe8).

> Note: we know that **snowflake_grant_privileges_to_role** doesn't work with ownership privilege right now.
> It's an expected behavior. To grant ownership we would like to have a new resource that would eventually
> replace existing **snowflake_role_ownership_grant** and **snowflake_user_ownership_grant**.

Any responses will be welcomed as they will have an impact on the overall grant ownership resource design and we will be able to incorporate mechanisms that would allow us to prevent some of the unpleasant cases from happening or make them less painful. Also, we would like to engage more in communication with you to build a product that will fit your needs and we would like to continue with initiatives as such in the future.

At the end of our grant discussions, the design decisions will be taken and we would like to produce a document that will be public and most likely will be placed in our GitHub repository. It should contain all of the things you need to know about grant ownership resources and we will try our best to inform you when it will be ready.

**Comment by PedroMartinSteenstrup** [comment]
@sfc-gh-jcieslak please open form to contributions, right now it's restricted to sf org I presume

**Reply by sfc-gh-jcieslak** [reply]
@PedroMartinSteenstrup My bad, It should work now

**Comment by sfc-gh-jcieslak** [comment]
Closing as the grant redesign was concluded. More info [here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1890#discussioncomment-9071073).
